python/cudf/cudf/utils/cudautils.py

```diff
@@ -77,12 +49,23 @@ def gpu_diff(in_col, out_col, out_mask, N):
 def gpu_mark_found_int(arr, val, out, not_found):
     i = cuda.grid(1)
     if i < arr.size:
-        if check_equals_int(arr[i], val):
+        if arr[i] == val:
             out[i] = i
         else:
             out[i] = not_found
 
 
+@jit
+def _check_equals_float(a, b):
+    """Check float equality with support for nans and infs."""
+    return (
+        a == b
+        or (math.isnan(a) and math.isnan(b))
+        or ((math.isinf(a) and a < 0) and (math.isinf(b) and b < 0))
+        or ((math.isinf(a) and a > 0) and (math.isinf(b) and b > 0))
```

Comment:
I think this was to workaround old Numba bugs where this can be simplified now to `a == b or (a != a and b != b)`